core/rawdb/accessors_state.go

```diff
@@ -47,10 +47,22 @@ func ReadCode(db ethdb.KeyValueReader, hash common.Hash) []byte {
 	if len(data) != 0 {
 		return data
 	}
-	data, _ = db.Get(hash[:])
+	data, _ = db.Get(hash.Bytes())
 	return data
 }
 
+// HasCode checks if the contract code corresponding to the
+// provided code hash is present in the db.
+func HasCode(db ethdb.KeyValueReader, hash common.Hash) bool {
```

Comment:
Can you move this function after `ReadCodeWithPrefix`? 
We have this order for db APIs `read, has, write, delete`. Would be nice to keep this order.